rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,624 @@
+    else:
+        molData = {}
+        
+    if 'viewInstantiated' in globals():
+        viewInstantiated = globals()['viewInstantiated']
+    else:
+        viewInstantiated = {}
+    
+    
+    viewInstantiated[uid] = False
+    
+    molViewState = MolViewState(uid, moldict, protein)
+    molData[uid] = molViewState
+    
+    keys = list(molViewState.moldict.keys())
+    
+    
+    # Right hand panel (widgets)
+    
+    wgBox=list()
+    
+    itemLayout=Layout(display='flex', flex_flow='row', justify_content='space-between')
+    
+    wgBox.append(Box([Label(value='uid'), HTML(description='', value=str(uid))], layout=itemLayout))
+    
+    wgData[uid]['selected_mols_view'] = HTML(description='', value=keys[0])
+    wgData[uid]['selected_confs_view'] = HTML(description='', value='0')
+    wgData[uid]['molId'] = Dropdown(description='', options=keys,value=keys[0])
+    wgData[uid]['selectMultiMols'] = Checkbox(description='selectMultiMols', value=False)
+    wgData[uid]['selectAllMols'] = Checkbox(description='selectAllMols', value=False)
+    wgData[uid]['confId'] = Dropdown(description='', options=list(range(9)), value=0)
+    wgData[uid]['selectMultiConfs'] = Checkbox(description='selectMultiConfs', value=False)
+    wgData[uid]['selectAllConfs'] = Checkbox(description='selectAllConfs', value=False)
+    
+    
+    wgBox.append(Box([Label(value='Mols'),wgData[uid]['selected_mols_view']], layout=itemLayout))
+    wgBox.append(Box([Label(value='Confs'),wgData[uid]['selected_confs_view']], layout=itemLayout))
+    
+    wgBox.append(Box([Label(value='molId'),wgData[uid]['molId']], layout=itemLayout))
+    
+    cbMolSelect=Box([Label(value=''),
+                     HBox([wgData[uid]['selectMultiMols'], 
+                           wgData[uid]['selectAllMols']])
+                    ], layout=itemLayout)
+    
+    wgBox.append(cbMolSelect)
+    
+    wgBox.append(Box([Label(value='confId'),wgData[uid]['confId']], layout=itemLayout))
+    
+    cbConfSelect=Box([Label(value=''),
+                      HBox([wgData[uid]['selectMultiConfs'], 
+                            wgData[uid]['selectAllConfs']])
+                     ], layout=itemLayout)
+    
+    wgBox.append(cbConfSelect)
+    
+    wgData[uid]['molId'].observe(handle_change, names='value')
+    wgData[uid]['selectMultiMols'].observe(handle_change, names='value')
+    wgData[uid]['selectAllMols'].observe(handle_change, names='value')
+    wgData[uid]['confId'].observe(handle_change, names='value')
+    wgData[uid]['selectMultiConfs'].observe(handle_change, names='value')
+    wgData[uid]['selectAllConfs'].observe(handle_change, names='value')
+    
+    wgMIds[wgData[uid]['molId'].model_id] = uid
+    wgMIds[wgData[uid]['selectMultiMols'].model_id] = uid
+    wgMIds[wgData[uid]['selectAllMols'].model_id] = uid
+    wgMIds[wgData[uid]['confId'].model_id] = uid
+    wgMIds[wgData[uid]['selectMultiConfs'].model_id] = uid
+    wgMIds[wgData[uid]['selectAllConfs'].model_id] = uid
```

Comment:
```
observers = ['molId', 'selectMultiMols', 'selectAllMols', 'confId',
             'selectMultiConfs', 'selectAllConds']
wgDataUid = wgData[uid]
for observerName in observers:
    observer = wgDataUid[observerName]
    observer.observe(handle_change, names='value')
    wgMIds[observer.model_id] = uid
```
Code might not be completely correct and would need to be moved further down.

There are more places where observers are linked to handle_change. It makes therefore sense to build the `observers` list in steps. I highlight a few places below.